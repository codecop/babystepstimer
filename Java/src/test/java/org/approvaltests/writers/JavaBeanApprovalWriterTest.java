package org.approvaltests.writers;

import org.approvaltests.Approvals;
import org.approvaltests.reporters.JunitReporter;
import org.approvaltests.reporters.QuietReporter;
import org.approvaltests.reporters.UseReporter;
import org.approvaltests.reporters.macosx.KDiff3Reporter;
import org.junit.Test;

import java.beans.IntrospectionException;
import java.lang.reflect.InvocationTargetException;
import java.util.Arrays;

public class JavaBeanApprovalWriterTest {

    String extractNoExceptions(Object o) {
        try {
            return JavaBeanApprovalWriter.extractData(o);
        } catch (IntrospectionException e) {
            e.printStackTrace();
        } catch (InvocationTargetException e) {
            e.printStackTrace();
        } catch (IllegalAccessException e) {
            e.printStackTrace();
        }
        return "";
    }

    @Test
    public void should_report_primitives() {
        Approvals.verifyAll("primitive", Arrays.<Object>asList(3, "3", 1.0), this::extractNoExceptions);
    }

    @Test
    public void should_report_arrays() {
        Approvals.verify(this.extractNoExceptions(new String[] { "xax", "xbx" }));
    }

    @Test
    //TODO should support primitive arrays
    public void should_ignore_primitive_arrays() {
        Approvals.verify(this.extractNoExceptions(new int[] { 1, 2 }));
    }

    @Test
    @UseReporter({ KDiff3Reporter.class, JunitReporter.class, QuietReporter.class })
    public void should_report_recurse_on_top_level_once_once() {
        Object[] recursiveDataStructure = { "a", null };
        recursiveDataStructure[1] = recursiveDataStructure;
        Approvals.verify(this.extractNoExceptions(recursiveDataStructure));
    }
}
