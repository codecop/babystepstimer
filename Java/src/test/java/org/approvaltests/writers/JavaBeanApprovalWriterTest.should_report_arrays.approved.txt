{[Ljava.lang.String;0 : {java.lang.String : xax
empty : {java.lang.Boolean : false
}
}
1 : {java.lang.String : xbx
empty : {java.lang.Boolean : false
}
}

}